hdlparse/vhdl_parser.py

```diff
@@ -172,19 +173,49 @@ def __init__(self, name, desc=None):
         self.desc = desc
 
 
+def remove_outer_parenthesis(s: Optional[str]):
+    if s:
+        n = 1
+        while n:
+            s, n = re.subn(r'\([^()]*\)', '', s.strip())  # remove non-nested/flat balanced parts
```

Comment:
Couple of things here:
* This regex seems to remove anything nested in parentheses which isn't what its name suggests.
* Parenthesis should be plural parentheses
* This function isn't called anywhere and isn't in a class?